### PR TITLE
Create skill

### PR DESCRIPTION
**Pull request by lukeaiken**
Association with Group.
Validations on Skill.
Create a Skill.

### CODE REVIEW

File: app/models/skill.rb [hunk 1]
```diff
@@ -1,4 +1,9 @@
 class Skill < ActiveRecord::Base
+  belongs_to :group
+
+  validates :name, presence: true
+  validates :featured, inclusion: [true, false]
```
Comment (ianderse) [inline app/models/skill.rb:5]: true and false are going to come through as strings, will that make a difference here?

**Comment by ianderse** [comment]
So it looks like a group is now a separate thing versus just a column on a skill? Why did this change? How does it pass through in the JSON now that it is separate?

**Comment by lukeaiken** [comment]
we never had group as a column. just pass in a group name. see controller

**Comment by lukeaiken** [comment]
the json formatting will take care of strings true and false

**Comment by okappati** [comment]
A group can exist independently of a skill, so it also doesn't make sense to have it be just a column in the skill table. I'm pretty sure we created a group table from the beginning. But just now created a join table. My opinion is that a skill should be able to have multiple group associations... would be good to check with @stevekinney tomorrow.

**Comment by ianderse** [comment]
I don't think a skill should have multiple groups. As was discussed before "Rails" skill for 1407 should be different from "Rails" skill for 1412. Although I suppose there could be edge cases, like "How happy are you at Turing?" skill for all groups. But then you lose the ability to see how an individual cohort responds.

**Comment by lukeaiken** [comment]
Skills only belong to one group

**Comment by ianderse** [comment]
See Kavita's comment.

**Comment by lukeaiken** [comment]
:)

**Pull request merged**